File: codec/elixir/lib/dcg_codec.ex
defmodule DcgCodec do
  def decode(deck_code) do
    DcgCodec.Decode.decode(deck_code)
  end

  def encode(deck, version \\ DcgCodec.Common.version()) do
    DcgCodec.Encode.encode(deck, version)
  end
end

File: codec/ruby/lib/dcg_codec.rb
# frozen_string_literal: true

require_relative 'dcg_codec/cli'
require_relative 'dcg_codec/decode'
require_relative 'dcg_codec/encode'

# DCGCodec module with encoder and decoder for Digimon Card Game (2020) deck codes
module DCGCodec
end

if $PROGRAM_NAME == __FILE__
  cli = DCGCodec::CLI.new
  cli.parse(ARGV)
end
